extensions/amp-base-carousel/1.0/amp-base-carousel.js

```diff
@@ -28,6 +28,21 @@ import {userAssert} from '../../../src/log';
 /** @const {string} */
 const TAG = 'amp-base-carousel';
 
+/**
+ * The boolean attribute value as resolved by string equality with "true" or "false".
+ * If the attribute is not present, default as given.
+ *
+ * @param {!Element} element
+ * @param {string} attr
+ * @param {boolean} defaultValue
+ * @return {boolean}
+ */
+function parseStrBoolAttr(element, attr, defaultValue) {
+  return element.hasAttribute(attr)
+    ? element.getAttribute(attr) !== 'false'
```

Comment:
Hmm. IMHO this should be a default boolean attribute interpretation logic for us in the `PractBaseElement`? WDYT? Btw, this is exactly how (P)react deal with booleans too, so I think it'd help us to be consistent.

Reply:
I go back and forth on this one. Normally I would agree to go with default boolean attribute logic, however this implies two pretty significant things for carousel:
1. It's simply not the same as 0.1, which looks for the string values, so something like an auto upgrade from 0.1 to 1.0 would be impossible.
2. Defaults become less intuitive. i.e. I would expect a default carousel without configurations to enable snapping to slides. This is true for `snap`, `horizontal` (which we could rename as `vertical` to make the logic work) and `loop`, which isn't currently aligned with 0.1 (default `true`) but happy to discuss also if it should be.

Reply:
Correction: [`loop` defaults to `false` in 0.1](https://github.com/ampproject/amphtml/issues/28661)

So really `snap` is the only one that would have an unintuitive default if we follow by presence.

Reply:
I definitely see your point on this. However, I'm not sure we actually have much of a problem. Let's assume our default boolean rules for the `{attr: 'y', default: true|false}` are:

1. `<x>`: we have to options: `y == default` or `y == undefined` and let component do its default.
2. `<x y>`: always `y == true`
3. `<x y="true">`: always `y == true`
4. `<x y="false">`: always `y == false`

And the implementation for this is pretty much exactly the code you have:
```
const value = el.hasAttribute(attr) ? el.getAttribute(attr) !== 'false'  : defaultValue;
```

These rules work very naturally for default-false cases such as `loop`. But I think they still work fine for default-true cases such as `snap`. It's just that the forms (2) and (3) are not strictly useful. But they are correct and there for completeness. 

WDYT?

Reply:
Sounds good to me, though I am not sure what I should do differently here other than restore `loop` to its original configuration as a boolean attribute in the classical sense of the term. It sounds like you are agreed that `snap` can stay as is, though we can inline the function for it since it will only be used there. Is that right?

Reply:
I think this would be needed:

1. Loop to go back to the old config.
2. Snap to switch to the similar config as loop, but perhaps a different default value.
3. [This code](https://github.com/ampproject/amphtml/blob/284e322f474cf84727c81dcf8bb77922f3398cde/src/preact/base-element.js#L721) should be modified to be more similar to the code you initial put together for this PR.

Reply:
Discussed offline -- Moved `loop` back to the old config, did not modify PreactBaseElement for now.